src/pages/md-asset-depreciation/form/AssetsForm.js

```diff
@@ -0,0 +1,324 @@
+    onSubmit: async obj => {
+      const formattedObj = {
+        ...obj,
+        date: formatDateToApi(obj.date)
+      }
+
+      const response = await postRequest({
+        extension: FixedAssetsRepository.AssetsDescription.set2,
+        record: JSON.stringify({ header: formattedObj, items: data.list || [] })
+      })
+
+      if (!obj.recordId) {
+        toast.success(platformLabels.Added)
+
+        formik.setFieldValue('recordId', response.recordId)
+        getData(response.recordId)
+      } else toast.success(platformLabels.Edited)
+      invalidate()
+    }
+  })
+
+  const editMode = !!formik.values.recordId
+
+  const onPost = async () => {
+    await postRequest({
+      extension: FixedAssetsRepository.AssetsDescription.post,
+      record: JSON.stringify(formik.values)
+    })
+
+    toast.success(platformLabels.Posted)
+    window.close()
+    invalidate()
+  }
+
+  useEffect(() => {
+    ;(async function () {
+      await getData(obj?.recordId)
+    })()
+  }, [])
+
+  const getData = async recordId => {
+    try {
+      if (recordId) {
+        const res = await getRequest({
+          extension: FixedAssetsRepository.AssetsDescription.get,
+          parameters: `_recordId=${recordId}`
+        })
+
+        formik.setValues({
+          ...res.record,
+          date: formatDateFromApi(res.record.date)
+        })
+      }
+    } catch (exception) {}
+  }
+
+  useEffect(() => {
+    ;(async function () {
+      if (obj?.recordId) {
+        const res = await getRequest({
+          extension: FixedAssetsRepository.AssetsTableData.qry,
+          parameters: `_depId=${obj.recordId}`
+        })
+        setData(res)
+      }
+    })()
+  }, [])
+
+  const columns = [
+    {
+      field: 'assetRef',
+      headerName: labels.asset,
+      flex: 1
+    },
+    {
+      field: 'assetName',
+      headerName: labels.name,
+      flex: 1
+    },
+    {
+      field: 'ccName',
+      headerName: labels.costCenter,
+      flex: 1
+    },
+    {
+      field: 'period',
+      headerName: labels.days,
+      flex: 1,
+      type: 'number'
+    },
+    {
+      field: 'amount',
+      headerName: labels.amount,
+      flex: 1
+    },
+    {
+      field: 'bookValue',
+      headerName: labels.bookValue,
+      flex: 1
+    },
+    {
+      field: 'depreciationAmount',
+      headerName: labels.depAmount,
+      flex: 1,
+      type: 'number'
+    },
+    {
+      field: 'newBookValue',
+      headerName: labels.newBookValue,
+      flex: 1
+    },
+    {
+      field: 'notes',
+      headerName: labels.notes,
+      flex: 1
+    }
+  ]
+
+  const actions = [
+    {
+      key: 'GL',
+      condition: true,
+      onClick: 'onClickGL',
+      disabled: !editMode
+    },
+    {
+      key: 'Post',
+      condition: true,
+      onClick: onPost,
+      disabled: !editMode || formik.values.status !== 1
+    },
+    {
+      key: 'PR',
+      condition: true,
+      onClick: fetchData,
+      disabled: editMode || preiewPressed
```

Comment:
this 'preiewPressed' will be a constant like editMode, that checks if formik.values.assets has data or not